### DISCUSSION TITLE
Type conversion precedence

### DISCUSSION THREAD

**Discussion by WesleyKapow** [josdejong/mathjs#2885] [Q&A] [answered]
I'm trying to be able to do both:
1. Add colors together, e.g. `"#F00" + "#0F0"` -> `"#FF0"`
2. Add strings together with a plus operator, e.g. `"hello " + user` -> `"Hello josdejong"`

I can get 1 working or 2 working but not both because the string concatenation is taking precedence over the color plus operator. Is there a way to specify which order newly added type operators will be tested/applied? Changing the order they are imported does not seem to have an affect.

Some code:
```ts
  const addString = mathjs.typed('add', {
    'string, string': function (a: string, b: string) {
      return a + b
    }
  })
  mathjs.import({ addString })

  mathjs.typed.addType({
    name: 'Color',
    test: function (maybeColor: string) {
      return maybeColor.startsWith("#")
    },
  })

  const addColor = mathjs.typed('add', {
    'Color, Color': function (a: string, b: string) {
      // some logic to add colors, as a test though we could just remove the `#` from the start of the second color.
      return a + b.slice(1)
    }
  })

  mathjs.import({ addColor })
```

**Comment by josdejong** [comment, marked as answer]
You'll indeed have to have the type check for `Color` before testing `string`, because all colors are also strings. Looking at [the API of `typed-function`](https://github.com/josdejong/typed-function), I see the `typed.addType` function doesn't allow this kind of ordering, but I think you can use `typed.addTypes(types, before)` where you can specify `before` as `"string"`.

**Reply by WesleyKapow** [reply]
After upgrading form version 10.x to 11.x, `addTypes` worked beautifully, thank you!!
